frappe/public/js/frappe/ui/messages.js

```diff
@@ -128,6 +128,19 @@ frappe.msgprint = function(msg, title, is_minimizable) {
 		data.indicator = 'blue';
 	}
 
+	if (data.as_list) {
+		const list_rows = data.message.map(m => `<li>${m}</li>`).join('');
+		data.message = `<ul style="padding-left: 20px">${list_rows}</ul>`;
+	}
+
+	if (data.as_table) {
+		const rows = data.message.map(row => {
+			const cols = row.map(col => `<td>${col}</td>`).join('');
+			return `<tr>${cols}</tr>`;
+		}).join('');
+		data.message = `<table class="table table-bordered" style="margin: 0;">${rows}</table>`;
+	}
+
 	if(data.message instanceof Array) {
```

Comment:
If this is not useful for anything then we can directly render a list

Reply:
If `frappe.msgprint` is called multiple times, it just appends the new message in the existing dialog. But it does renders the title of the last message. To handle this better, we should show all the messages with their titles in the dialog body.